File: python3/poor_pigs.py
"""
Space   : O(1)
Time    : O(log N)
"""


class Solution:
    def poorPigs(self, buckets: int, minutesToDie: int, minutesToTest: int) -> int:
        t = (minutesToTest // minutesToDie) + 1

        x = 0
        while t**x < buckets:
            x += 1

        return x

File: golang/poor_pigs.go
package main

/*
Space	: O(1)
Time	: O(log N)
*/

import "math"

func poorPigs(buckets int, minutesToDie int, minutesToTest int) int {
    t := (minutesToTest / minutesToDie) + 1
    
    x := 0
    for int(math.Pow(float64(t), float64(x))) < buckets {
        x++
    }
    
    return x
}
